<launch>

<arg name="map_size_x" default="20.0"/>
<arg name="map_size_y" default="20.0"/>
<arg name="map_size_z" default=" 5.0"/>

<arg name="init_x" default="-8.0"/>
<arg name="init_y" default="-8.0"/>
<arg name="init_z" default=" 2.0"/>

  <node pkg="teach_repeat_planer" type="tr_node" name="tr_node" output="screen" required = "true">
      <remap from="~waypoints"   to="/waypoint_generator/waypoints"/>
      <remap from="~odometry"    to="/drone_1/state_ukf/odom"/>
      <remap from="~map"         to="/random_complex/global_map"/> 
      <!-- <remap from="~command"     to="/position_cmd_input"/>  -->
      <remap from="~joystick"    to="/joy"/> 
      
      <param name="optimization/poly_order"  value="8"     /> 
      <param name="optimization/min_order"   value="3.0"   /> 
      <param name="optimization/objective"   value="energy"/> 
      <param name="optimization/mode"        value="teach"/> 
      
      <param name="map/x_size"       value="$(arg map_size_x)"/>
      <param name="map/y_size"       value="$(arg map_size_y)"/>
      <param name="map/z_size"       value="$(arg map_size_z)"/>
      
      <param name="map/margin"       value="0.1" />
      <param name="map/resolution"   value="0.1" />
      
      <param name="planning/max_inflate"   value="100"  />
      <param name="planning/step_length"   value="1"    />
      <param name="planning/cube_margin"   value="0.0"  />

      <param name="planning/rho_time"      value="20.0"  />
      <param name="planning/max_vel"       value="4.0"  />
      <param name="planning/max_acc"       value="3.0"  />
      <param name="planning/max_d_acc"     value="3.0" />

      <param name="vis/vis_traj_width"     value="0.15" />
      <param name="vis/vis_man_path_width" value="0.10" />
      <param name="vis/is_proj_cube"       value="true" />
      <param name="vis/corridor_trans"     value="0.6" />
  </node>

  <!-- <node pkg  = "teach_repeat_planer" type = "tr_traj_server" output = "screen" name = "tr_traj_server">    
      <remap from = "~trajectory"        to = "/tr_node/trajectory"/>
      <remap from = "~odometry"          to = "/drone_1/state_ukf/odom"/>
      <remap from = "~position_command"  to = "/position_cmd_test"/>
  </node> -->

  <node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
      <remap from="~odom" to="/drone_1/state_ukf/odom"/>        
      <remap from="~goal" to="/goal"/>
      <param name="waypoint_type" value="manual-lonely-waypoint"/>    
  </node>

  <node pkg ="teach_repeat_planer" name ="random_complex" type ="random_complex" output = "screen">    
      <remap from="~odometry"       to="/drone_1/state_ukf/odom"/>

      <param name="init_state_x"   value="$(arg init_x)"/>
      <param name="init_state_y"   value="$(arg init_y)"/>

      <param name="map/x_size"     value="$(arg map_size_x)" />
      <param name="map/y_size"     value="$(arg map_size_y)" />
      <param name="map/z_size"     value="$(arg map_size_z)" />

      <param name="map/circle_num" value="50"/>        
      <param name="map/obs_num"    value="200"/>        
      <param name="map/resolution" value="0.1"/>        

      <param name="ObstacleShape/lower_rad" value="0.1"/>
      <param name="ObstacleShape/upper_rad" value="0.8"/>
      <param name="ObstacleShape/lower_hei" value="1.0"/>
      <param name="ObstacleShape/upper_hei" value="3.0"/>

      <param name="CircleShape/lower_circle_rad"   value="0.6"/>        
      <param name="CircleShape/upper_circle_rad"   value="2.0"/>        
      
      <param name="sensing/rate"   value="10.0"/>        
  </node>

<!--   <node pkg="odom_visualization" name="odom_visualization_ukf_" type="odom_visualization" output="screen">
      <remap from="~odom" to="/odom/fake_odom"/>
      <param name="color/a" value="0.8"/>    
      <param name="color/r" value="0.0"/>        
      <param name="color/g" value="0.0"/>        
      <param name="color/b" value="0.0"/>       
      <param name="covariance_scale" value="100.0"/>       
      <param name="robot_scale" value="1.0"/>   
  </node>
 -->
</launch>